indexer/services/comlink/src/controllers/api/v4/affiliates-controller.ts

```diff
@@ -116,38 +119,65 @@ class AffiliatesController extends Controller {
 
   @Get('/snapshot')
   async getSnapshot(
-    @Query() offset?: number,
+    @Query() addressFilter?: string[],
+      @Query() offset?: number,
       @Query() limit?: number,
-      @Query() sortByReferredFees?: boolean,
+      @Query() sortByAffiliateEarning?: boolean,
   ): Promise<AffiliateSnapshotResponse> {
-    const finalOffset = offset ?? 0;
-    const finalLimit = limit ?? 1000;
-    // eslint-disable-next-line
-    const finalSortByReferredFees = sortByReferredFees ?? false;
+    const finalAddressFilter: string[] = addressFilter ?? [];
+    const finalOffset: number = offset ?? 0;
+    const finalLimit: number = limit ?? 1000;
+    const finalsortByAffiliateEarning: boolean = sortByAffiliateEarning ?? false;
 
-    // simulate a delay
-    await new Promise((resolve) => setTimeout(resolve, 100));
+    // eslint-disable-next-line max-len
+    const infos: AffiliateInfoFromDatabase[] | undefined = await AffiliateInfoTable.paginatedFindWithAddressFilter(
+      finalAddressFilter,
+      finalOffset,
+      finalLimit,
+      finalsortByAffiliateEarning,
+    );
 
-    const snapshot: AffiliateSnapshotResponseObject = {
-      affiliateAddress: 'some_address',
-      affiliateReferralCode: 'TempCode123',
-      affiliateEarnings: 100,
-      affiliateReferredTrades: 1000,
-      affiliateTotalReferredFees: 100,
-      affiliateReferredUsers: 10,
-      affiliateReferredNetProtocolEarnings: 1000,
-      affiliateReferredTotalVolume: 1000000,
-    };
+    // No results found
+    if (infos === undefined) {
+      return {
+        affiliateList: [],
+        total: 0,
+        currentOffset: finalOffset,
+      };
+    }
 
-    const affiliateSnapshots: AffiliateSnapshotResponseObject[] = [];
-    for (let i = 0; i < finalLimit; i++) {
-      affiliateSnapshots.push(snapshot);
+    // Get referral codes
+    // eslint-disable-next-line max-len
+    const addressUsernames: AddressUsernameFromDatabase[] = await SubaccountUsernamesTable.findByAddress(
```

Comment:
```suggestion
    const addressUsernames:
    AddressUsernameFromDatabase[] = await SubaccountUsernamesTable.findByAddress(
```
then remove the eslint-dsiable-next-line